tests/unit/normalize_test.py

```diff
@@ -0,0 +1,176 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2015, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+"""Tests normalization scheme for normalizing detectors' scores according
+to baseline."""
+
+import csv
+import json
+import os
+import shutil
+import tempfile
+import unittest
+
+from nab.runner import Runner
+
+
+def createTemporaryCsv(parentDir, fileName, data):
+  """Write a temporary CSV file for testing.
+
+  @param parentDir a tempfile temporary directory
+  @param fileName string name of file
+  @param data list of lists to write to CSV
+  """
+  filePath = os.path.join(parentDir, fileName)
+
+  with open(filePath, 'w') as csvfile:
+    filewriter = csv.writer(csvfile, delimiter=',')
+    for ln in data:
+      filewriter.writerow(ln)
+
+  return filePath
+
+
+def createRunner(resultsDir, profileName, resultsName=None):
+  """Create a nab.runner.Runner object for testing."""
+  testRunner = Runner(dataDir=None,
+                      resultsDir=resultsDir,
+                      labelPath=None,
+                      profilesPath=None,
+                      thresholdPath=None)
+
+  testRunner.profiles = json.loads('{"'+profileName+'": {}}')
+
+  if resultsName is not None:
+    resultsFile = resultsName+'_'+profileName+'_scores.csv'
+    resultsFilePath = os.path.join(resultsDir, resultsName, resultsFile)
+    testRunner.resultsFiles = [resultsFilePath]
+
+  return testRunner
+
+
+
+class NormalizationTest(unittest.TestCase):
+
+  def setUp(self):
+    self._tmpDirs = [] # stores absolute paths of dirs created during testing
```

Comment:
`setUp()` and `tearDown()` are called for each test method, so this is unnecessary. If a future test creates multiple temp dirs we'll modify these methods then.